<launch>
  <node
    pkg="perception"
    type="webcam_pub.py"
    name="webcam_pub"
    output="screen"
  />
  <node
    pkg="perception"
    type="workspace_feature_detection.py"
    name="workspace_feature_detection"
    output="screen"
  />
  <node
    pkg="trajectory_generation"
    type="robot_trajectory_generation.py"
    name="robot_trajectory_generation"
    output="screen"
  />
  <node
    pkg="control"
    type="mission_control.py"
    name="mission_control"
    output="screen"
  />
  <node
    pkg="control"
    type="IKService.py"
    name="ik_service"
    output="screen"
  />
  <node
    pkg="control"
    type="FKService.py"
    name="fk_service"
    output="screen"
  />
  <node name="joint1_pid_service" pkg="control" type="PIDService.py">
    <remap from="pid_set_gains" to="joint1_pid_set_gains"/>
    <remap from="pid_set_output_limits" to="joint1_pid_set_output_limits"/>
    <remap from="pid_compute" to="joint1_pid_compute"/>
  </node>
  <node name="joint2_pid_service" pkg="control" type="PIDService.py">
    <remap from="pid_set_gains" to="joint2_pid_set_gains"/>
    <remap from="pid_set_output_limits" to="joint2_pid_set_output_limits"/>
    <remap from="pid_compute" to="joint2_pid_compute"/>
  </node>
  <node name="joint3_pid_service" pkg="control" type="PIDService.py">
    <remap from="pid_set_gains" to="joint3_pid_set_gains"/>
    <remap from="pid_set_output_limits" to="joint3_pid_set_output_limits"/>
    <remap from="pid_compute" to="joint3_pid_compute"/>
  </node>
  <node
    pkg="motor"
    type="motor_read_write.py"
    name="read_write_py_node"
    output="screen"
  /> 
</launch>